src/server/middleware.ts

```diff
@@ -0,0 +1,42 @@
+import type { NextFunction, Response } from "express";
+import { UserService } from "../shared/services";
+
+export function setDefaultCsp({
+  res,
+  next,
+}: {
+  res: Response;
+  next: NextFunction;
+}) {
+  res.setHeader(
+    "Content-Security-Policy",
+    `default-src 'self'; manifest-src *; connect-src *; img-src * data:; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; form-action 'self'; base-uri 'self'; frame-src *; media-src *`
```

Comment:
Are these really necessary?

`unsafe-inline` and `unsafe-eval` introduce serious security risks.

Please avoid usage or use with nonce

Reply:
Afaik they are required by the js framework, would be great if we could get rid of them.

Reply:
@pixlguru you can try removing these as a separate PR, but make sure you do real testing on it.

Reply:
Maybe I could do this after 18.1 is out, if I remember correctly ,last time I took a look unsave-eval was used by websocket code, so there is a chance that removing it has no side-effects.